### PR TITLE
Claim social media id achievement

### PR DESCRIPTION
**Pull request by matt-goldman**

### CODE REVIEW

File: src/SSW.Rewards.Application/Achievements/Command/ClaimSocialMediaAchievementForUser/ClaimSocialMediaAchievementForUser.cs [hunk 1]
```diff
@@ -23,24 +23,27 @@ public async Task<int> Handle(ClaimSocialMediaAchievementForUser request, Cancel
...
             userAchievement = new UserAchievement
             {
-                AchievementId = achievementId.Value,
-                UserId        = currentUserId,
-                AwardedAt     = _dateTimeService.Now,
-                CreatedUtc    = _dateTimeService.Now
+                Achievement = achievement,
+                User        = user, 
+                AwardedAt   = _dateTimeService.Now,
+                CreatedUtc  = _dateTimeService.Now
```
Comment (lukecookssw) [inline src/SSW.Rewards.Application/Achievements/Command/ClaimSocialMediaAchievementForUser/ClaimSocialMediaAchievementForUser.cs:46]: You don't need to get this user object, or the achievement object. 2 unnecessary db round-trips.

Just set the userId and achievementId you already have.

File: src/SSW.Rewards.Application/Users/Commands/UpsertUserSocialMediaId/UpsertUserSocialMediaId.cs [hunk 2]
```diff
@@ -28,9 +28,11 @@ public async Task<bool> Handle(UpsertUserSocialMediaId request, CancellationToke
     {
         bool isInsert = false;
 
-        var platform = await _context.SocialMediaPlatforms.FindAsync(request.SocialMediaPlatformId, cancellationToken);
+        var platform = await _context.SocialMediaPlatforms
+            .FirstOrDefaultAsync(p => p.AchievementId == request.AchievementId, cancellationToken);
+
```
Comment (lukecookssw) [inline src/SSW.Rewards.Application/Users/Commands/UpsertUserSocialMediaId/UpsertUserSocialMediaId.cs:33]: AsNoTracking()

**Review by lukecookssw — CHANGES_REQUESTED** [review]
couple of changes

File: src/SSW.Rewards.Application/Achievements/Command/ClaimSocialMediaAchievementForUser/ClaimSocialMediaAchievementForUser.cs (hunk 1 above)
Comment (matt-goldman) [reply src/SSW.Rewards.Application/Achievements/Command/ClaimSocialMediaAchievementForUser/ClaimSocialMediaAchievementForUser.cs:46]: That's what I was originally doing and it caused an unexpected issue. It wouldn't record this entity in the db, and added a new user to the users table with no details. I changed it to work like this and it solved the problem.
I will add a comment saying why this is done this wat and a PBI to investigate.

File: src/SSW.Rewards.Application/Users/Commands/UpsertUserSocialMediaId/UpsertUserSocialMediaId.cs (hunk 2 above)
Comment (matt-goldman) [reply src/SSW.Rewards.Application/Users/Commands/UpsertUserSocialMediaId/UpsertUserSocialMediaId.cs:33]: Done

File: src/SSW.Rewards.Application/Achievements/Command/ClaimSocialMediaAchievementForUser/ClaimSocialMediaAchievementForUser.cs (hunk 1 above)
Comment (matt-goldman) [reply src/SSW.Rewards.Application/Achievements/Command/ClaimSocialMediaAchievementForUser/ClaimSocialMediaAchievementForUser.cs:51]: Done

**Review by lukecookssw — APPROVED** [review]

**Pull request merged**